core/history/history-xwiki/src/components/componentsInit.ts

```diff
@@ -0,0 +1,171 @@
+    if (documentId == null) {
+      this.alertsService.error(
+        `Could not load page history for ${pageData.name}.`,
+      );
+      return [];
+    }
+
+    const restApiSearchParams = new URLSearchParams();
+    if (limit) {
+      restApiSearchParams.append("number", limit.toString());
+    }
+    if (offset) {
+      restApiSearchParams.append("start", offset.toString());
+    }
+    const restApiUrl =
+      getRestSpacesApiUrl(this.cristalApp.getWikiConfig(), documentId) +
+      "/history";
+
+    try {
+      const authorization = await this.authenticationManagerProvider
+        .get()
+        ?.getAuthorizationHeader();
+      const headers: { Accept: string; Authorization?: string } = {
+        Accept: "application/json",
+      };
+
+      if (authorization) {
+        headers.Authorization = authorization;
+      }
+      const response = await fetch(restApiUrl, { headers });
+      const jsonResponse = await response.json();
+      const revisions: Array<PageRevision> = await Promise.all(
+        jsonResponse.historySummaries.map(
+          async (revision: {
+            version: string;
+            modified: string;
+            modifier: string;
+            comment: string;
+          }) => {
+            return {
+              version: revision.version,
+              date: new Date(revision.modified),
+              user: {
+                profile: this.cristalApp.getRouter().resolve({
+                  name: "view",
+                  params: {
+                    page: revision.modifier,
+                  },
+                }).href,
+                name: await this.getUserName(revision.modifier),
+              },
+              comment: revision.comment,
+              url: this.cristalApp.getRouter().resolve({
+                name: "view",
+                params: {
+                  page: documentId,
+                  revision: revision.version,
+                },
+              }).href,
+            };
+          },
+        ),
+      );
+      return revisions;
+    } catch (error) {
+      this.logger.error(error);
+      this.alertsService.error(
+        `Could not load page history for ${pageData.name}.`,
+      );
+      return [];
+    }
+  }
+
+  private async getUserName(userId: string): Promise<string> {
+    const restApiUrl =
+      getRestSpacesApiUrl(this.cristalApp.getWikiConfig(), userId) +
+      "/objects/XWiki.XWikiUsers/0";
+
+    try {
+      const authorization = await this.authenticationManagerProvider
+        .get()
+        ?.getAuthorizationHeader();
+      const headers: { Accept: string; Authorization?: string } = {
+        Accept: "application/json",
+      };
+
+      if (authorization) {
+        headers.Authorization = authorization;
+      }
+      const response = await fetch(restApiUrl, { headers });
+      const jsonResponse = await response.json();
+      let user = jsonResponse.pageName;
+      jsonResponse.property.forEach(
```

Comment:
Just to be 100% sure, shouldn't it be plural (i.e., properties)?
```suggestion
      jsonResponse.properties.forEach(
```

Reply:
No, I checked
![image](https://github.com/user-attachments/assets/c5c22240-bd6d-4c73-a551-345b59fc4336)